cmd/restic/cmd_check.go

```diff
@@ -355,13 +372,14 @@ func runCheck(ctx context.Context, opts CheckOptions, gopts GlobalOptions, args
 	doReadData := func(packs map[restic.ID]int64) {
 		packCount := uint64(len(packs))
 
-		p := newTerminalProgressMax(!gopts.Quiet, packCount, "packs", term)
+		p := newTerminalProgressMax(!gopts.Quiet && !gopts.JSON, packCount, "packs", term)
```

Comment:
```suggestion
		p := printer.NewCounter("packs")
		p.SetMax(uint64(len(packs)))
```

Reply:
passing the maximum value to `newTerminalProgressMax` or setting it via `SetMax` is equivalent

Reply:
I only now noticed the `p.SetMax()` call and fixed it. Done.